pkg/odo/cli/component/watch.go

```diff
@@ -74,6 +74,14 @@ var watchCmd = &cobra.Command{
 		}
 		watchPath := util.ReadFilePath(u, runtime.GOOS)
 
+		rules, err := util.GetIgnoreRulesFromDirectory(watchPath)
+		if err != nil {
+			odoutil.CheckError(err, "")
+		}
+		if len(rules) > 0 {
+			ignores = append(ignores, rules...)
```

Comment:
Isn't this going to concatenate rules provided as flag and rules from files?
If `--ignore` flag is provided then `odoignore` and `gitignore` should be ignored